web/components/Feed/Feed.tsx

```diff
@@ -299,18 +342,48 @@ function Feed(props: {
         } else {
           tempState[filt.key].push(filt.options[ind]);
         }
-        console.log(tempState);
-        return tempState;
+        toReturn = tempState;
+        break;
       default:
-        return state;
+        toReturn = state;
     }
+    setQueryFilters(getQueryFilters(toReturn));
+    return toReturn;
   }
 
+  const [queryFilters, setQueryFilters] = useState(
+    getQueryFilters(getInitialFilters())
+  );
```

Comment:
The selectedFilters object is laid out for the frontend to be able to easily display the filters but it needs to be converted into the format required for the filter for the api endpoint

Reply:
Got it. I think in this case, since `queryFilters` is derived from `selectedFilters`, you could do something with `feedPosts` to eliminate the intermediate state `queryFilters`:

```js
const feedPosts = trpc.post.getFilteredPosts
    .useQuery(getQueryFilters(selectedFilters))
    ?.data?.map((p: IPost) => {
      return {
        attachments: p.attachments,
        date: p.date,
        name: p.name,
        tag: p.type,
        description: p.description,
      };
    });
```